src/EnergyPlus/Pipes.hh

```diff
@@ -1,129 +1,53 @@
 		int BranchIndex; // Index of plant Branch index where this pipe resides
 		int CompIndex; // Index of plant Comp index where this pipe resides
-		bool OneTimeInit;
-		bool CheckEquipName;
-		bool EnvrnFlag;
 
 		// Default Constructor
 		LocalPipeData() :
-			TypeOf( 0 ),
 			InletNodeNum( 0 ),
 			OutletNodeNum( 0 ),
 			LoopNum( 0 ),
 			LoopSide( 0 ),
 			BranchIndex( 0 ),
-			CompIndex( 0 ),
-			OneTimeInit( true ),
-			CheckEquipName( true ),
-			EnvrnFlag( true )
+			CompIndex( 0 )
 		{}
 
-		// Member Constructor
-		LocalPipeData(
-			std::string const & Name, // main plant (cooling) loop ID
-			int const TypeOf, // type of pipe
-			int const InletNodeNum, // Node number on the inlet side of the plant
-			int const OutletNodeNum, // Node number on the inlet side of the plant
-			int const LoopNum, // Index of plant loop where this pipe resides
-			int const LoopSide, // Index of plant loop side where this pipe resides
-			int const BranchIndex, // Index of plant Branch index where this pipe resides
-			int const CompIndex, // Index of plant Comp index where this pipe resides
-			bool const OneTimeInit,
-			bool const CheckEquipName,
-			bool const EnvrnFlag
-		) :
-			Name( Name ),
-			TypeOf( TypeOf ),
-			InletNodeNum( InletNodeNum ),
-			OutletNodeNum( OutletNodeNum ),
-			LoopNum( LoopNum ),
-			LoopSide( LoopSide ),
-			BranchIndex( BranchIndex ),
-			CompIndex( CompIndex ),
-			OneTimeInit( OneTimeInit ),
-			CheckEquipName( CheckEquipName ),
-			EnvrnFlag( EnvrnFlag )
-		{}
+		static std::shared_ptr<PlantComponent> pipeFactory( std::string objectName );
+		int performEveryTimeInit();
+		int performOneTimeInit();
+		int performBeginEnvrnInit();
+		int simulate();
 
 	};
 
 	// Object Data
-	extern Array1D< LocalPipeData > LocalPipe; // dimension to number of pipes
-
-	// Functions
-
-	void
-	SimPipes(
-		int const CompType,
-		std::string & PipeName,
-		int & CompIndex,
-		Real64 const MaxVolFlowRate,
-		bool const InitLoopEquip,
-		bool const FirstHVACIteration
-	);
-
-	// End Plant Loop Module Driver Subroutines
-	//******************************************************************************
-
-	// Beginning of Plant Loop Module Get Input subroutines
-	//******************************************************************************
-
-	void
-	GetPipeInput();
-
-	// End of Get Input subroutines for the Plant Loop Module
-	//******************************************************************************
-
-	// Beginning Initialization Section of the Plant Loop Module
-	//******************************************************************************
-
-	void
-	InitializePipes(
-		int const PipeType, // Type of Pipe
-		std::string const & PipeName, // Name of Pipe
-		int & PipeNum, // Index into pipe structure for name
-		Real64 const MaxVolFlowRate // unused at present time
-	);
+	extern Array1D< std::shared_ptr< LocalPipeData > > LocalPipe; // dimension to number of pipes
```

Comment:
@jasondegraw This is a good debate to have now (or a year ago ;-). I think it is hard to work on a code that is partly 0-based and partly 1-based and, while you may be comfortable with 0-based, I have worked with groups transitioning from Fortran and 0-based is a big source of grief (and bugs). But, happily, the collection framework, like the Arrays, does both:
```
collection(1); // First element
collection[0]; // First element
```
Yes, this can be another source of confusion without good communication, but it does let you go with 1-based for now and switch down the road. And it would take about 3 minutes to make the collections do everything 0-based, so let's get this debated and decided!